src/sdk/object_sdk.js

```diff
@@ -245,18 +256,20 @@ class ObjectSDK {
         }
         // check for a specific bucket
         if (bucket && req.op_name !== 'put_bucket') {
-            // ANONYMOUS: cannot work without bucket, cannot work on namespace bucket (?)
+            // ANONYMOUS: cannot work without bucket.
+            // Return if the acount is anonymous
+            const is_anon = !(token && token.access_key);
```

Comment:
this causing s3 requests with access key empty and secret key non empty to make anon requests instead of failing like happens in AWS